labels/labels.go

```diff
@@ -117,5 +118,30 @@ func Shorten(input, pkg string) string {
 // takes care of the optional ":" prefix and differences between long-form
 // labels and local labels (relative to pkg).
 func Equal(label1, label2, pkg string) bool {
-	return ParseRelative(label1, pkg) == ParseRelative(label2, pkg)
+	parsedLabel1, parsedLabel2 := ParseRelative(label1, pkg), ParseRelative(label2, pkg)
+	if parsedLabel1 == parsedLabel2 {
+		return true
+	}
+
+	// cover the case when the first labels package is fully contained in the second
```

Comment:
line 127-135 is an extremely aggressive strategy. If you include this it might result in getting rid of used proto deps that you might have to add back in later.